go-controller/pkg/metrics/ovs.go

```diff
@@ -684,6 +695,64 @@ func setOvsInterfaceStatusFields(interfaceBridge, interfacePort, interfaceName,
 		interfaceName, firmwareVersion).Set(1)
 }
 
+func getGeneveInterfaceStatsFieldValue(stats *netlink.LinkStatistics, field string) float64 {
+	r := reflect.ValueOf(stats)
+	fieldValue := reflect.Indirect(r).FieldByName(field)
+	return float64(fieldValue.Uint())
+}
+
+func setGeneveInterfaceStatistics(geneveInterfaceName string, link netlink.Link) {
+	var geneveInterfaceStatsMap = map[string]string{
+		"rx_packets":   "RxPackets",
+		"rx_bytes":     "RxBytes",
+		"rx_dropped":   "RxDropped",
+		"rx_frame_err": "RxFrameErrors",
+		"rx_over_err":  "RxOverErrors",
+		"rx_crc_err":   "RxCrcErrors",
+		"rx_errors":    "RxErrors",
+		"tx_packets":   "TxPackets",
+		"tx_bytes":     "TxBytes",
+		"tx_dropped":   "TxDropped",
+		"collisions":   "Collisions",
+		"tx_errors":    "TxErrors",
+	}
+
+	for statsName, geneveStatsName := range geneveInterfaceStatsMap {
+		metricName := "interface_" + statsName
+		metricValue := getGeneveInterfaceStatsFieldValue(link.Attrs().Statistics, geneveStatsName)
+		ovsInterfaceMetricsDataMap[metricName].metric.WithLabelValues(
+			"none", "none", geneveInterfaceName).Set(metricValue)
+	}
+}
+
+// geneveInterfaceMetricsUpdate updates the geneve interface
+// metrics obtained through netlink library equivalent to
+// (ip -s li show genev_sys_6081)
+func geneveInterfaceMetricsUpdate() (err error) {
+
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
@Pardhakeswar why do we need a recover here? what situations would cause a panic?

Reply:
yeah, we don't need it in this context.

Reply:
@dcbw @girishmg updated the PR by removing the recover code.